chroma_core/models/stratagem.py

```diff
@@ -105,41 +105,113 @@ def service_file(fid):
 
 
 class ConfigureStratagemTimerStep(Step, CommandLine):
+    def calc_duration(self, duration, units):
+        if units is None:
+            return str(duration)
+        elif units in ("s", "seconds"):
+            return "{}s".format(duration / 1000)
+
+    def get_run_stratagem_command(self, cmd, config, units):
+        iml_cmd = "{} --filesystem {}".format(cmd, config.filesystem.id)
+        if config.report_duration is not None and config.report_duration >= 0:
+            duration = self.calc_duration(config.report_duration, units)
+            iml_cmd += " --report {}".format(duration)
+        if config.purge_duration is not None and config.purge_duration >= 0:
+            duration = self.calc_duration(config.purge_duration, units)
+            iml_cmd += " --purge {}".format(duration)
+
+        return iml_cmd
+
     def run(self, kwargs):
         job_log.debug("Configure stratagem timer step kwargs: {}".format(kwargs))
         # Create systemd timer
 
         config = kwargs["config"]
 
-        with open(timer_file(config.id), "w") as fn:
-            fn.write(
-                "#  This file is part of IML\n"
-                "#  This file will be overwritten automatically\n"
-                "\n[Unit]\n"
-                "Description=Start Stratagem run on {}\n"
-                "\n[Timer]\n"
-                "OnActiveSec={}\n"
-                "OnUnitActiveSec={}\n".format(config.filesystem.id, config.interval / 1000, config.interval / 1000)
+        if runningInDocker():
+            iml_cmd = self.get_run_stratagem_command("/usr/bin/start-stratagem-scan", config, None)
+
+            # Create timer file
+            timer_config = """# This file is part of IML
+# This file will be overwritten automatically
+
+[Unit]
+Description=Start Stratagem run on {}
+
+[Timer]
+OnActiveSec={}
+OnUnitActiveSec={}
```

Comment:
Nit: maybe use named parameters for readability and robustness. As a side effect, can dedup `config.interval / 1000`

Reply:
sure 👍